### PR TITLE
Project completed, applied for review

### PR DESCRIPTION
**Pull request by GzimAsani**
- This project was created using Javascript, HTML/CSS, and Bootstrap
- This project allows users to create new projects.
- It allows users to create new tasks but to create a new task user has to select a project.
- The task lists are created with different background colors depending on the priority of the task.
- It allows users to edit and delete tasks.

### CODE REVIEW

**Comment by GzimAsani** [comment]
We got a changes requested in dashboard but no comment is showing here! :confused: . I am applying for another code review

**Review by descholar-ceo — CHANGES_REQUESTED** [review]
# Status: Changes Required :vertical_traffic_light: :warning: 
## Hello @GzimAsani and @Div685  :handshake: :handshake: :handshake: 

:+1: This is good to make it this far, your project is really very nice, congratulations on the good work you have done so far.
:+1: I liked that you have tried to implement all minimal requirements of this project, keep it up, and your project is really almost done.
#### :warning: However we need to make few changes to mark this project approved. See the requested change below :point_down: :point_down: :point_down: 
### Note: All of requested changes which are marked as [Required] are mandatory to be implemented in order to get this project approved, others which are marked as [Optional] are nice to have in your project to make it outstanding

# [Required Changes] :red_circle: 
- [x] Files not related to the project should not be committed and added to .gitignore (e.g. .vscode) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/gitignore.png) from Pathwright) => Remove the `.vscode` folder from your project
- [x] Add some validations so that a user should be asked to fill duedate if they forgot to fill them up. Currently you are validating only the task title but not for date, hence the task created without due date would not make sense
- [x] Keep all the styling logic in CSS files (If you need to change something like a background image, add the classes with the different backgrounds to the CSS and then change the class name attribute with JavaScript) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/section/styling_requirement.png) from Pathwright).  For example instead of writing this `taskDivcolor[i].style.background = '#ed1250';` you can switch classes, then that class contains the backound you want to switch to.
- [x] Because to run the project locally we have to follow this step `Open your favorite browser to open the index.html file located in the dist folder.` Then there is no need to install dependecies, since everything is bundled when this file `dist/main.js` is being made by webpack 

# [Optional changes] :yellow_circle: 
- [ ] Add a custom error message so that if a user create a task without selecting a project, they would be displayed why it is failed and what they need to do. Currently when you do such thing, it's not showing anything but it consoles an error, so it would be very good if you take care of that!
- [x] It will be even more splendid if you add an active class to the currently selected project
- [ ] It would be very nice if there is a way of deleting a project
___
_If anything seems to be unclear to you, reachout and ask anything by commenting on this Pull Request and mention me as [descholar-ceo](https://github.com/descholar-ceo), I will receive a notification and will respond._

[GitHub](https://github.com/descholar-ceo) | [LinkedIn](https://linkedin.com/in/mugirase-emmanuel) | [Twitter](https://twitter.com/descholar3)

_Reviewed by [Mugirase Emmanuel (descholar)](https://github.com/descholar-ceo)_

**Review by mikenath223 — CHANGES_REQUESTED** [review]
### Hello Team 👋

**You've done pretty well here fixing a few issues raised in the previous review. 👌**

However, let's make a couple of changes.

- Please refer to comments on the code.

### **STATUS: CHANGES REQUESTED** 👷‍♂️

Kindly fix these issues and request another review👍
Feel free to contact me on slack(michgolden) if you need further clarification.

**HAPPY CODING 🎉💻🎉**

**Michgolden Ukeje (CR)**

File: src/color.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  const taskDivcolor = [...document.querySelectorAll('.task')];
+  for (let i = 0; i < todos.length; i += 1) {
+    for (let i = 0; i < selectedList.tasks.length; i += 1) {
+      if (selectedList.tasks[i].priority === 'High') {
+        taskDivcolor[i].style.background = '#ed1250';
+      } else if (selectedList.tasks[i].priority === 'Medium') {
+        taskDivcolor[i].style.background = '#d3d00f';
+      } else {
+        taskDivcolor[i].style.background = '#0fc53d';
+      }
```
Comment (mikenath223) [inline src/color.js]: - [x] As stated in the previous review kindly move all CSS styling to the stylesheet then toggle the styles of the DOM element by toggling the classes of the elements. Please do this for **all** styles in the js modules.

File: src/domElement.js [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+      listElement.classList.add('active-list');
+    }
+    listsContainer.appendChild(listElement);
+  });
+}
+
+function editTask(task, label) {
+  editTaskForm();
+  newTaskInput.value = task.name;
+  newTaskDate.value = task.date;
```
Comment (mikenath223) [inline src/domElement.js]: - [x] Kindly change these and **all** other regular functions to es6 arrow function syntax.

**Review by SpaYco — APPROVED** [review]
# Project Approved

thank you for making the required changes, your project is now approved!

### please don't forget to merge to master, good luck with the next projects!


Best, 
Aziz Mejri 
_contact me on [twitter](https://twitter.com/iSpaYco) or slack (@SpaYco) if you need any help_

**Pull request merged**